internal/indexcoord/garbage_collector.go

```diff
@@ -138,9 +137,18 @@ func (gc *garbageCollector) recycleUnusedIndexFiles() {
 				if !gc.metaTable.CanBeRecycledIndexFiles(buildID) {
```

Comment:
seems that no need to implement CanBeRecycledIndexFiles.

GetIndexFilePathInfo should have exact same logic of CanBeRecycledIndexFiles?

I notice CanBeRecycledIndexFiles check meta.indexMeta.State != commonpb.IndexState_Finished || meta.indexMeta.State != commonpb.IndexState_FAILED

should we also check FAILED in GetIndexFilePathInfo

Reply:
I think so, it will be better. Done.